setenv function conflicts with gflags

Hello, 
I tried to compile gflags with Mingw. But as soon as I compile with mingw32-make. 

> C:\ceres\gflags-bin>mingw32-make
> Scanning dependencies of target gflags_static
> [ 10%] Building CXX object CMakeFiles/gflags_static.dir/src/gflags.cc.obj
> In file included from C:/ceres/gflags-bin/include/gflags/config.h:110:0,
>                  from C:\ceres\gflags-master\src\gflags.cc:90:
> C:/ceres/gflags-master/src/windows_port.h: In function 'void setenv(const char*, const char*, int)':
> C:/ceres/gflags-master/src/windows_port.h:80:13: error: ambiguating new declaration of 'void setenv(const char*, const char*, int)'
>  inline void setenv(const char* name, const char* value, int) {
>              ^~~~~~
> In file included from c:\mingw\lib\gcc\mingw32\6.3.0\include\c++\cstdlib:75:0,
>                  from c:\mingw\lib\gcc\mingw32\6.3.0\include\c++\stdlib.h:36,
>                  from C:/ceres/gflags-master/src/windows_port.h:56,
>                  from C:/ceres/gflags-bin/include/gflags/config.h:110,
>                  from C:\ceres\gflags-master\src\gflags.cc:90:
> c:\mingw\include\stdlib.h:837:30: note: old declaration 'int setenv(const char*, const char*, int)'
>  __cdecl __MINGW_NOTHROW  int setenv( const char *__n, const char *__v, int __f )
>                               ^~~~~~
> CMakeFiles\gflags_static.dir\build.make:62: recipe for target 'CMakeFiles/gflags_static.dir/src/gflags.cc.obj' failed
> mingw32-make[2]: *** [CMakeFiles/gflags_static.dir/src/gflags.cc.obj] Error 1
> CMakeFiles\Makefile2:66: recipe for target 'CMakeFiles/gflags_static.dir/all' failed
> mingw32-make[1]: *** [CMakeFiles/gflags_static.dir/all] Error 2
> Makefile:150: recipe for target 'all' failed
> mingw32-make: *** [all] Error 2


I also tried building with/without shared. I don't know how remove this issue. Is it impossible to compile with mingw? 

System information

Operating System => Windows10 64 Bit
Compiler => Mingw32_6.3
Cmake => 3.9.0-rc3

--- a/src/gflags.cc
+++ b/src/gflags.cc
@@ -1036,16 +1036,15 @@ uint32 CommandLineFlagParser::ParseNewCommandLineFlags(int* argc, char*** argv,
     char* arg = (*argv)[i];
 
     // Like getopt(), we permute non-option flags to be at the end.
-    if (arg[0] != '-' || arg[1] == '\0') {	// must be a program argument "-" is an argument, not a flag
+    if (arg[0] != '-' || arg[1] == '\0') {	// must be a program argument: "-" is an argument, not a flag
       memmove((*argv) + i, (*argv) + i+1, (*argc - (i+1)) * sizeof((*argv)[i]));
       (*argv)[*argc-1] = arg;      // we go last
       first_nonopt--;              // we've been pushed onto the stack
       i--;                         // to undo the i++ in the loop
       continue;
     }
-
-    if (arg[0] == '-') arg++;      // allow leading '-'
-    if (arg[0] == '-') arg++;      // or leading '--'
+    arg++;                     // skip leading '-'
+    if (arg[0] == '-') arg++;  // or leading '--'
 
     // -- alone means what it does for GNU: stop options parsing
     if (*arg == '\0') {

--- a/src/gflags_completions.cc
+++ b/src/gflags_completions.cc
@@ -550,8 +550,7 @@ static void FinalizeCompletionOutput(
 
   vector<DisplayInfoGroup> output_groups;
   bool perfect_match_found = false;
-  if (lines_so_far < max_desired_lines &&
-      !notable_flags->perfect_match_flag.empty()) {
+  if (!notable_flags->perfect_match_flag.empty()) {
     perfect_match_found = true;
     DisplayInfoGroup group =
         { "",

